### ISSUE TITLE
ClassNotFoundException: org.signal.libsignal.net.NetworkException

### ISSUE THREAD

**Issue by Terrance** [signalapp/Signal-Android#13516]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

This was originally reported against a fork (johanw666#98), but it looks like this may be an upstream issue.

----------------------------------------

### Bug description

Signal is repeatedly crashing with the following:

```
FATAL EXCEPTION: Thread-10
Process: org.thoughtcrime.securesms, PID: 3480
java.lang.ClassNotFoundException: org.signal.libsignal.net.NetworkException
```

I use [Rethink](https://github.com/celzero/rethink-app) as a firewall on Android, and it looks like the crashes happen when a network request to `cdsi.signal.org` fails (perhaps this is a new endpoint added in upstream 7.4.0.0).  Having allowed the domain, the app seems to have stopped crashing, but blocking it again doesn't break things -- looks like it's a one-off request, but another one is made a day or two later.

The bit quoted above came from the crash log with no other context.  The relevant part of the main log, still unfortunately lacking any kind of stack trace:

```
[7.4.0.0-JW] [287674] 2024-04-11 15:43:53.146 GMT+01:00 I CdsiV2Service: Starting CDSI lookup via libsignal-net
[7.4.0.0-JW] [287672] 2024-04-11 15:43:53.148 GMT+01:00 I JobRunner: [JOB::d0420fc7-7d6a-4de9-907c-029fad4ef534][RotateCertificateJob][1] Job finished with result SUCCESS in 491 ms. (Time Since Submission: 499 ms, Lifespan: 86400000 ms, Run Attempt: 1/Unlimited, Queue: __ROTATE_SENDER_CERTIFICATE__)
[7.4.0.0-JW] [287698] 2024-04-11 15:43:56.580 GMT+01:00 I libsignal: rust/net/src/infra/connection_manager.rs:193: Connection attempt failed with an error: transport: Failed to establish TCP connection to any of the IPs
[7.4.0.0-JW] [287690] 2024-04-11 15:44:00.922 GMT+01:00 I WebSocketConnection: [normal:33720672] Sending keep alive...
[7.4.0.0-JW] [287690] 2024-04-11 15:44:00.926 GMT+01:00 I WebSocketConnection: [unidentified:167615400] Sending keep alive...
[7.4.0.0-JW] [287690] 2024-04-11 15:44:00.935 GMT+01:00 D AlarmSleepTimer: Setting an exact alarm to wake up in 19995ms.
[7.4.0.0-JW] [287699] 2024-04-11 15:44:01.611 GMT+01:00 I libsignal: rust/net/src/infra/connection_manager.rs:193: Connection attempt failed with an error: transport: Failed to establish TCP connection to any of the IPs
[7.4.0.0-JW] [287700] 2024-04-11 15:44:01.656 GMT+01:00 I libsignal: rust/net/src/infra/connection_manager.rs:193: Connection attempt failed with an error: transport: Failed to establish TCP connection to any of the IPs
[7.4.0.0-JW] [287701] 2024-04-11 15:44:03.161 GMT+01:00 E SignalUncaughtException: 
[7.4.0.0-JW] [287701] 2024-04-11 15:44:03.161 GMT+01:00 E SignalUncaughtException: java.lang.ClassNotFoundException: org.signal.libsignal.net.NetworkException
[7.4.0.0-JW] [287701] 2024-04-11 15:44:03.161 GMT+01:00 E SignalUncaughtException: 
[7.4.0.0-JW] [287701] 2024-04-11 15:44:03.231 GMT+01:00 I JobManager: Successfully flushed.
```

### Steps to reproduce

- open Signal
- wait a few seconds

**Actual result:** Signal crashes
**Expected result:** Signal stays open, potentially reporting its inability to connect to services

### Device info

**Device:** OnePlus Nord CE
**Android version:** 13
**Signal version:** 7.4.0.0-JW (based on upstream tag 7.4.0)

### Link to debug log

📎 [log.txt](https://github.com/signalapp/Signal-Android/files/14962522/log.txt)

**Comment by Terrance** [comment]
Looks like this may have been fixed by https://github.com/signalapp/Signal-Android/commit/b3336b4d84def3c167b5fd115d1b5395cb06c65f?  v7.4.2 appears not to crash under the same conditions.

**Comment by clark-signal** [comment]
Yes, should be fixed in 7.4.2! Thanks for the report
